taskiq/scheduler/scheduler.py

```diff
@@ -56,3 +59,17 @@ async def startup(self) -> None:  # pragma: no cover
         connections or anything you'd like.
         """
         await self.broker.startup()
+
+    async def on_ready(self, task: ScheduledTask) -> None:
+        """
+        This method is called on task is ready to be enqueued.
+
+        It's triggered on proper time depending on `task.cron` or `task.time` attribute.
+        :param task: task to send
+        """
+        await task.source.pre_send(task)
```

Comment:
Can you wrap it in `maybe_awaitable`? This will add you posibility to add sync functions.

```python
	if task.source.pre_send != TaskiqScheduler.pre_send:
		await maybe_awaitable(task.source.pre_send(task))
```